### ISSUE TITLE
Need to find a way to delete deployments on devnet automatically

### ISSUE THREAD

**Issue by sabrinasadik** [threefoldtech/home#1314]
Devnet seems to have a lot of deployments that were needed for testing but never got removed. This fills up resources of the devnet nodes that can be used for other deployments. 

Is there a way to nuke everything automatically after an x-period of time?

**Comment by xmonader** [comment]
I believe devnet can be nuked in general :D

**Comment by coesensbert** [comment]
> I believe devnet can be nuked in general :D

yeah but then the issue will come back at some point. Same with public IP's, they are all mostly always reserved.

**Comment by xmonader** [comment]
@muhamadazmy @LeeSmet please follow up here

**Comment by LeeSmet** [comment]
There's a couple of options here:

- Reset devnet chain entirely. This means devnet chain will be restored to the genesis state, all accounts will be empty, all contracts gone. There are no events on chain, so the nodes will probably not understand what is going on, hence they will need to be wiped and rebooted to. This would be a hassle.
- Add some code which periodically nukes workloads on chain. This can be automated, but will be a hassle to properly implement and get performant. This also introduces environment specific code, which can impact testing of the other code. Lastly, this might be enabled on other networks by accident, which would be a rather unfortunate issue.
- - Add some code which terminates contracts after x amount of time maximum on devnet. Issues similar to the above.
- Add an extrinisc only callable by sudo, which kills all contracts. This would lead to proper cleanup as events would be emitted, so the nodes don't need to do anything (assuming contract cleanup works). This does mean the extrinsic is available on test/mainnet and can be called there by accident. Also by mushing this all into 1 extrinsic, it might become too heavy for the block if there is a really large amount of contracts, killing the chain (shouldn't happen but still).
- Use the sudo key to do a `call as` (i.e. masquerade as) contract owners and cancel the contract. With a script which loads all contracts and their owners, we can create and submit an extrinsic to terminate the contract while pretending to be the owner. Nodes get events and thus clean up as required. We also have control over which contracts to terminate if needed.

IMO, the latter seems to be the best solution as it is both the cleanest way to terminate contracts (node's don't need a reboot), the easiest to implement (just create a script to grab contracts and their owners and make the call to cancel), and the least impactful on the code base (no actual code changes).
